File: decompiled/sources/com/hjq/http/lifecycle/HttpLifecycleManager.java
package com.hjq.http.lifecycle;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleEventObserver;
import androidx.lifecycle.LifecycleOwner;
import com.hjq.http.EasyHttp;

public final class HttpLifecycleManager implements LifecycleEventObserver {
    public static void bind(LifecycleOwner lifecycleOwner) {
        lifecycleOwner.getLifecycle().addObserver(new HttpLifecycleManager());
    }

    public static boolean isLifecycleActive(LifecycleOwner lifecycleOwner) {
        return (lifecycleOwner == null || lifecycleOwner.getLifecycle().getCurrentState() == Lifecycle.State.DESTROYED) ? false : true;
    }

    public void onStateChanged(LifecycleOwner lifecycleOwner, Lifecycle.Event event) {
        if (event == Lifecycle.Event.ON_DESTROY) {
            lifecycleOwner.getLifecycle().removeObserver(this);
            EasyHttp.cancel(lifecycleOwner);
        }
    }
}

File: decompiled/sources/com/hjq/http/lifecycle/HttpLifecycleManager.kt
package com.hjq.http.lifecycle

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleEventObserver
import androidx.lifecycle.LifecycleOwner
import com.hjq.http.EasyHttp

class HttpLifecycleManager : LifecycleEventObserver {
    companion object {
        @JvmStatic
        fun bind(lifecycleOwner: LifecycleOwner) {
            lifecycleOwner.lifecycle.addObserver(HttpLifecycleManager())
        }

        @JvmStatic
        fun isLifecycleActive(lifecycleOwner: LifecycleOwner?): Boolean {
            return lifecycleOwner != null && lifecycleOwner.lifecycle.currentState != Lifecycle.State.DESTROYED
        }
    }

    override fun onStateChanged(lifecycleOwner: LifecycleOwner, event: Lifecycle.Event) {
        if (event == Lifecycle.Event.ON_DESTROY) {
            lifecycleOwner.lifecycle.removeObserver(this)
            EasyHttp.cancel(lifecycleOwner)
        }
    }
}
